chainerx_cc/chainerx/backward.cc

```diff
@@ -590,39 +597,67 @@ std::vector<nonstd::optional<Array>> Grad(
         const std::vector<ConstArrayRef>& outputs,
         const std::vector<ConstArrayRef>& inputs,
         const nonstd::optional<BackpropId>& backprop_id,
-        DoubleBackpropOption double_backprop) {
+        DoubleBackpropOption double_backprop,
+        bool set_grad,
+        bool retain_grad,
+        const std::vector<ConstArrayRef>& grad_inputs,
+        const std::vector<ConstArrayRef>& grad_outputs) {
     if (inputs.empty()) {
         return {};
     }
     if (outputs.empty()) {
         return std::vector<nonstd::optional<Array>>(inputs.size(), nonstd::nullopt);
     }
 
+    if (grad_inputs.size() && (inputs.size() != grad_inputs.size())) {
```

Comment:
```suggestion
    if (!grad_inputs.empty() && inputs.size() != grad_inputs.size()) {
```

* Parentheses are redundant
* Do not evaluate a non-bool value as bool.